src/toast/types.ts

```diff
@@ -1,20 +1,28 @@
-import React from "react";
+import React, { HTMLAttributes } from "react";
 
 export type ToastType = "success" | "warning" | "error" | "info";
 
-export interface ToastProps extends React.HTMLAttributes<HTMLDivElement> {
+export interface ActionButtonProps {
```

Comment:
will need a more unique name to avoid conflicts with other components

thinking if this can directly inherit `ButtonProps` instead

```suggestion
export interface ToastActionButtonProps {
```